ttnn/cpp/ttnn/operations/eltwise/binary/common/binary_op_utils.cpp

```diff
@@ -164,7 +164,7 @@ std::map<std::string, std::string> get_defines(
 
     if (input_tensor_a_activation.has_value()) {
         defines.merge(ttnn::operations::unary::utils::get_defines(
-            input_tensor_a_activation.value().op_type, std::nullopt, "PRE_IN0_0", idst));
+            input_tensor_a_activation.value().op_type, std::nullopt, "PRE_IN0_0", idst, input_dtype));
```

Comment:
This was because I attempted to fuse the `GTZ` and `LEZ` unary ops and got an error because the dtype wasn't passed in. While this isn't strictly necessary for the implementation of `where` that does not use fused ops (because it failed for other reasons) it is a relevant bugfix.